doc/radosgw: Promptify CLI, cosmetic fixes

bluikko opened the pull request:
Use the more modern prompt block for CLI commands and use right one $ vs #.

Fix indentation on JSON example outputs and some CLI command switches.

Add some arguably missing comma in JSON example output.

Add a full stop at the end of a one-sentence paragraph. Remove extra comma mid-sentence in another.

Fix missing backslashes or typo at end of multiline commands.

Lines under section headings as long as heading text.

Fix hyperlinks. Fix list items prefixed with `-` insted of `*`. Format configuration syntax in the middle of text as ``code``.

Fix typo "PI" to "API" and remove extra space between it and the following full stop. Remove colons at the end of section headers in a few places. Use Title Case in section titles consistently with short words lowercase.

Don't try to create code blocks with triple backtick as it won't work, use double colon (in the end of preceding text paragraph as other files do, could be just `::` on its own line too I suppose).

Possibly controversial: don't add whitespace before and after main title section header text. Few were like this and changed for consistency with other such title lines.

Possibly controversial: don't indent line continuation backslashes, leave only 1 space before them.





<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test classic perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-classic/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test crimson perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-crimson/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test signed` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pr-commits/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-commits/config/definitions/ceph-pr-commits.yml)
- `jenkins test make check` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests/config/definitions/ceph-pull-requests.yml)
- `jenkins test make check arm64` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests-arm64/config/definitions/ceph-pull-requests-arm64.yml)
- `jenkins test submodules` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-pr-submodules/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-submodules/config/definitions/ceph-pr-commits.yml)
- `jenkins test dashboard` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml)
- `jenkins test dashboard cephadm` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-cephadm-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-cephadm-e2e/config/definitions/ceph-dashboard-cephadm-e2e.yml)
- `jenkins test api` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-api/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-api/config/definitions/ceph-pr-api.yml)
- `jenkins test docs` [ReadTheDocs](https://readthedocs.org/projects/ceph/) | [Github Workflow Definition](https://github.com/ceph/ceph/blob/main/.readthedocs.yml)
- `jenkins test ceph-volume all` [Jenkins Jobs](https://jenkins.ceph.com/view/ceph-volume%20PR/) | [Jenkins Jobs Definition](https://github.com/ceph/ceph-build/blob/main/ceph-volume-cephadm-prs/config/definitions/ceph-volume-pr.yml)
- `jenkins test windows` [Jenkins Job](https://jenkins.ceph.com/job/ceph-windows-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-windows-pull-requests/config/definitions/ceph-windows-pull-requests.yml)
- `jenkins test rook e2e` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-orchestrator-rook-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-rook-e2e/config/definitions/ceph-orchestrator-rook-e2e.yml)
</details>

ceph-jenkins commented:
Can one of the admins verify this patch?

anthonyeleven commented on doc/radosgw/cloud-restore.rst:
```diff
@@ -12,23 +12,22 @@ S3-compatible cloud services. In order to faciliate this,
 --------------------------------------
 
-The `tier configuration <https://docs.ceph.com/en/latest/radosgw/cloud-transition/#cloud-storage-class-configuration>`
+The `tier configuration <https://docs.ceph.com/en/latest/radosgw/cloud-transition/#cloud-storage-class-configuration>`_
```
What is the importance of the trailing underscore?

bluikko replied:
Without the trailing underscore it is not recognized as a hyperlink, it's all rendered in text but just italic. Check out https://docs.ceph.com/en/latest/radosgw/cloud-restore/ vs https://ceph--62763.org.readthedocs.build/en/62763/radosgw/cloud-restore/

bluikko replied:
Regarding this particular link in your OP. I just realized that links inside Ceph documentation surely must not be hardcoded URLs and they should be links to a RST reference instead?

A grep of `doc/` seems to confirm this, there are very very few hyperlinks pointing to `https://docs.ceph.com/`.

anthonyeleven replied:
>This is how this whole PR became a rabbit hole: every revision I noticed masses of more things to "improve" (some arguably very minor or opinionated).

Welcome to my world :D

We call this "making a clean spot"

https://www.youtube.com/watch?v=qMmzIp5ubqM

>Without the trailing underscore it is not recognized as a hyperlink, it's all rendered in text but just italic.

Ack. 

>A grep of doc/ seems to confirm this, there are very very few hyperlinks pointing to https://docs.ceph.com/.
`
Really there shouldn't be any, we should be using `:ref:` perhaps, but it seems to work without that.  For sure we should not be using explicit URLs to refer to other parts of our own documentation, as the FQDN may change over time, and we may render into different forms, so please do feel free to update any links of the above form.

anthonyeleven commented on doc/radosgw/cloud-restore.rst:
```diff
@@ -12,23 +12,22 @@ S3-compatible cloud services. In order to faciliate this,
+The `tier configuration <https://docs.ceph.com/en/latest/radosgw/cloud-transition/#cloud-storage-class-configuration>`_
 of the cloud storage class configured for data transition is used to restore
-objects as well:
+objects as well::
```
Why double colon?

bluikko replied:
It will render the following block as a code block. This is used in some other files. The triple backtick is just rendered as-is and does not create a code block like in markdown.

bluikko replied:
Ah, right, I forgot to list this change in the PR / commit description sorry for that.

bluikko replied:
Oof, more of these both (link missing underscore and triple backtick) also in `cloud-transition.rst` but I think this PR is fat enough as is. Will leave it for tomorrow if that's fine.  
Added the double colon in PR description.

anthonyeleven replied:
In most cases I think we use a double backtick; I don't recall a triple anywhere but that doesn't mean it doesn't happen.  Remember that we use Sphinx vs the Markdown one might see on GitHub, sometimes there are rendering subtleties.  If you have an improved strategy for code / command blocks, I'm all ears, I'll ask though that you lay out the distinctions for me.

bluikko replied:
I see I forgot to check `cloud-transition.rst` as rendered -- this PR did touch that file, too. I can either update this PR tomorrow (out of time now) or do a different PR, which do you prefer?

anthonyeleven replied:
updating this one would be my pref.  Some of the lines you touch are ones that I very recently edited myself.

bluikko replied:
Double backtick seems to be for code inline, not a code block?

It would seem double colon is the syntax to use for a code block. Is the question then whether that double colon should be on its own line by itself, or can it be merged to the preceding single colon at the end of the previous text paragraph?

So either something like:
```
text paragraph here. For example::

    code block
```

or

```
text paragraph here. For example:

::

    code block
```

I also do see sometimes
```
.. code-block::

    code-block
```
The double colon at the end of a text paragraph renders as a single colon & creates a code block. It seems to be used at least in `doc/ceph-volume/` and `doc/cephadm/`; also the third example seems to be used in `doc/cephdm/`.

I guess it's a matter of personal preference or policy how to do this.

anthonyeleven replied:
Hmmm:

>The "code" directive constructs a literal block. If the code language is specified, the content is parsed by the [Pygments](https://pygments.org/) syntax highlighter and tokens are stored in nested [inline elements](https://docutils.sourceforge.io/docs/ref/doctree.html#inline-elements) with class arguments according to their syntactic category.

and

>A paragraph consisting of two colons (::) signifies that the following text block(s) comprise a literal block. The literal block must either be indented or quoted (see below). No markup processing is done within a literal block. It is left as-is, and is typically rendered in a monospaced typeface:

I think when we deal with classic Markdown, Slack Markdown, GitHub Markdown, and RST on the same day it's all too easy to mix up the contexts.

Looking into this I think we should probably be using the

```
text paragraph here. For example:

::

    code block

```
form.  Compared to triple backticks I think this renders with a shaded background.

bluikko commented:
Fixed more links and markdownisms in `cloud-transition.rst`, also fixed indentation in multiline promptified CLI commands. Will confirm it's all fixed after the PR rendering pipeline job is complete.

bluikko commented:
OK `cloud-transitions.rst` has the same fixes done as other files.  
Indenting multiline commands had problems I had created and was missing some of my fixes after the initial commit. I couldn't just leave it like that.

This PR is "really, really ready" now and should have all rendered docs now checked including `cloud-transitions.rst` that I missed earlier.

anthonyeleven commented:
Indeed links should be relative.  On Apr 11, 2025, at 12:52 AM, Ville Ojamo ***@***.***> wrote:﻿
@bluikko commented on this pull request.



In doc/radosgw/cloud-restore.rst:
>  API can be used to restore the object temporarily.
 
 
 Cloud Storage Class Tier Configuration
 --------------------------------------
 
-The `tier configuration <https://docs.ceph.com/en/latest/radosgw/cloud-transition/#cloud-storage-class-configuration>`
+The `tier configuration <https://docs.ceph.com/en/latest/radosgw/cloud-transition/#cloud-storage-class-configuration>`_

Regarding this particular link in your OP. I just realized that links inside Ceph documentation surely must not be hardcoded URLs and they should be links to a RST reference instead?
A grep of doc/ seems to confirm this, there are very very few hyperlinks pointing to https://docs.ceph.com/.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because your review was requested.Message ID: ***@***.***>

bluikko commented:
First of all thank you for bearing with me.  
If you would like to take this discussion to some other forum I'm happy to do that, I can create Slack account for example so that we can have a more quick exchange.

I have gone through all the 6 files changed by this PR and confirmed that the current state of the PR could be mergeable with no new errors caused by this PR and the rendered documents looking as I intended them.  
If you have some specific change requests or feedback I will address it ASAP.

Meanwhile, respectfully, I would like to discuss hyperlinks in some more detail. I might be misunderstanding what exactly you mean with "relative" links; I was trying to point out that any such "hardcoded" link might be discouraged whether absolute or relative and RST references might be preferable.  
This is underlined by the fact that the "hardcoded" link that prompted this discussion in https://github.com/ceph/ceph/pull/62763#discussion_r2037882763 is actually a broken link, the anchor does not exist and following the link results in top of the document instead of the relevant section/anchor `cloud-storage-class-configuration`.

Please note that I am by no means an expert in RST, I may get terminology and details wrong but I'm basing this mostly on the Sphinx documentation below and how the files in `doc/` usually do things.

> Indeed links should be relative.

I meant RST references like the ones listed here in the documentation: https://www.sphinx-doc.org/en/master/usage/referencing.html#cross-referencing-arbitrary-locations

So there are a few different ways to create hyperlinks, I will list here 3 using RST features and 1 using hardcoded URLs (whether absolute or relative).  
Using RST references will create automatic link labels, for example if I add a reference just preceding a section with title `Multisite Configuration` then the link to that reference will automatically get the correct link text/label of "Multisite Configuration".  
Or I could directly link to a section using the section title text, without adding a reference in the destination document but this would break anchors in links if the section title is ever changed.

1) Linking to an RST reference when a specific section in the middle a document needs to be linked (like in this case the intended rendered link ends in an anchor), example:
   ```
   See `Multisite Configuration`_ for multisite configuration instructions.
   ```
   This `Multisite Configuration` could be an RST item such as section title text or a custom defined anchor such as:
   ```
   .. _Multisite Configuration: ../multisite
   ```

2) Linking to an RST reference similarly as above but with different syntax, I believe this is about the same as the option (1) above:
   ```
   :ref:`ceph-volume-lvm`
   ```
   And the reference that the hyperlink points to is defined similarly as same as above option (1). For example here the above link to reference `ceph-volume-lvm` will automatically get the link text/label from the section title ("`lvm`"):
   ```
   doc/ceph-volume/lvm/index.rst:.. _ceph-volume-lvm:
   doc/ceph-volume/lvm/index.rst-
   doc/ceph-volume/lvm/index.rst-``lvm``
   doc/ceph-volume/lvm/index.rst-=======
   ```

3) Linking a document name without an anchor in the intended link, in effect linking to the beginning of the document:
   ```
   :doc:`cloud-transition`
   ```

4) I believe this would be discouraged, a hardcoded link using HTML feature to an URL whether relative or absolute:
   ```
   `upmap <../rados/operations/upmap>`_
   ```

Some statistics from the `doc/` code base:
- 28k RST references of the above variety (1).
- 631 RST references of the above variety (2)
- 162 RST references of the above variety (3).
- 120 hardcoded relative links of the above variety (4)  
   Almost all of them point to "special" content such as man pages and changelog .txt files, and not to RST files.
- Only 15 hardcoded absolute links of the above variety (4) to docs.ceph.com.

These statistics point to _RST references being the preferred method of linking other RST content_, which seems logical. These are metadata inside the documentation and not tied to structure on the file system or the resulting URLs as rendered for the web server that may be changed over time resulting in broken links (I believe these are not linted to detect such cases??).

```
$ fgrep '`_' -R doc/ |wc -l
28767
$ fgrep ':ref:`' -R doc/ |wc -l
631
$ fgrep ':doc:`' -R doc/ |wc -l
162
$ fgrep '<../' -R doc/ |wc -l
120
$ fgrep '<https://docs.ceph.com' -R doc/ |wc -l
15
```

anthonyeleven commented:
I’ll follow up in more detail this weekend.

I suggest joining the Ceph Slack workspace : https://join.slack.com/t/ceph-storage/shared_invite/zt-32uc4bnb5-rEWQpR5cdGIjmdyUozY8iw

Where there is a #ceph-doc channel

Best to have discussions whee others can contribute.


> On Apr 11, 2025, at 4:20 AM, Ville Ojamo ***@***.***> wrote:
> 
> 
> bluikko
>  left a comment 
> (ceph/ceph#62763)
> First of all thank you for bearing with me.
> If you would like to take this discussion to some other forum I'm happy to do that, I can create Slack account for example so that we can have a more quick exchange.
> 
> I have gone through all the 6 files changed by this PR and confirmed that the current state of the PR could be mergeable with no new errors caused by this PR and the rendered documents looking as I intended them.
> If you have some specific change requests or feedback I will address it ASAP.
> 
> Meanwhile, respectfully, I would like to discuss hyperlinks in some more detail. I might be misunderstanding what exactly you mean with "relative" links; I was trying to point out that any such "hardcoded" link might be discouraged whether absolute or relative and RST references might be preferable.
> This is underlined by the fact that the "hardcoded" link that prompted this discussion in #62763 (comment) is actually a broken link, the anchor does not exist and following the link results in top of the document instead of the relevant section/anchor cloud-storage-class-configuration.
> 
> Please note that I am by no means an expert in RST, I may get terminology and details wrong but I'm basing this mostly on the Sphinx documentation below and how the files in doc/ usually do things.
> 
> Indeed links should be relative.
> 
> I meant RST references like the ones listed here in the documentation: https://www.sphinx-doc.org/en/master/usage/referencing.html#cross-referencing-arbitrary-locations
> 
> So there are a few different ways to create hyperlinks, I will list here 3 using RST features and 1 using hardcoded URLs (whether absolute or relative).
> Using RST references will create automatic link labels, for example if I add a reference just preceding a section with title Multisite Configuration then the link to that reference will automatically get the correct link text/label of "Multisite Configuration".
> Or I could directly link to a section using the section title text, without adding a reference in the destination document but this would break anchors in links if the section title is ever changed.
> 
> Linking to an RST reference when a specific section in the middle a document needs to be linked (like in this case the intended rendered link ends in an anchor), example:
> 
> See `Multisite Configuration`_ for multisite configuration instructions.
> This Multisite Configuration could be an RST item such as section title text or a custom defined anchor such as:
> 
> .. _Multisite Configuration: ../multisite
> Linking to an RST reference similarly as above but with different syntax, I believe this is about the same as the option (1) above:
> 
> :ref:`ceph-volume-lvm`
> And the reference that the hyperlink points to is defined similarly as same as above option (1). For example here the above link to reference ceph-volume-lvm will automatically get the link text/label from the section title ("lvm"):
> 
> doc/ceph-volume/lvm/index.rst:.. _ceph-volume-lvm:
> doc/ceph-volume/lvm/index.rst-
> doc/ceph-volume/lvm/index.rst-``lvm``
> doc/ceph-volume/lvm/index.rst-=======
> Linking a document name without an anchor in the intended link, in effect linking to the beginning of the document:
> 
> :doc:`cloud-transition`
> I believe this would be discouraged, a hardcoded link using HTML feature to an URL whether relative or absolute:
> 
> `upmap <../rados/operations/upmap>`_
> Some statistics from the doc/ code base:
> 
> 28k RST references of the above variety (1).
> 631 RST references of the above variety (2)
> 162 RST references of the above variety (3).
> 120 hardcoded relative links of the above variety (4)
> Almost all of them point to "special" content such as man pages and changelog .txt files, and not to RST files.
> Only 15 hardcoded absolute links of the above variety (4) to docs.ceph.com.
> These statistics point to RST references being the preferred method of linking other RST content, which seems logical. These are metadata inside the documentation and not tied to structure on the file system or the resulting URLs as rendered for the web server that may be changed over time resulting in broken links (I believe these are not linted to detect such cases??).
> 
> $ fgrep '`_' -R doc/ |wc -l
> 28767
> $ fgrep ':ref:`' -R doc/ |wc -l
> 631
> $ fgrep ':doc:`' -R doc/ |wc -l
> 162
> $ fgrep '<../' -R doc/ |wc -l
> 120
> $ fgrep '<https://docs.ceph.com' -R doc/ |wc -l
> 15
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because your review was requested.
> 
>  <https://github.com/ceph/ceph/pull/62763#discussion_r2037882763> <https://www.sphinx-doc.org/en/master/usage/referencing.html#cross-referencing-arbitrary-locations> <https://github.com/ceph/ceph/pull/62763#issuecomment-2796187940> <https://github.com/notifications/unsubscribe-auth/ADTVONEI3E4UPG3EKVN4XYT2Y53LJAVCNFSM6AAAAAB23JJ2HSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJWGE4DOOJUGA>
> 
> bluikko
>  left a comment 
> (ceph/ceph#62763)
>  <https://github.com/ceph/ceph/pull/62763#issuecomment-2796187940>
> First of all thank you for bearing with me.
> If you would like to take this discussion to some other forum I'm happy to do that, I can create Slack account for example so that we can have a more quick exchange.
> 
> I have gone through all the 6 files changed by this PR and confirmed that the current state of the PR could be mergeable with no new errors caused by this PR and the rendered documents looking as I intended them.
> If you have some specific change requests or feedback I will address it ASAP.
> 
> Meanwhile, respectfully, I would like to discuss hyperlinks in some more detail. I might be misunderstanding what exactly you mean with "relative" links; I was trying to point out that any such "hardcoded" link might be discouraged whether absolute or relative and RST references might be preferable.
> This is underlined by the fact that the "hardcoded" link that prompted this discussion in #62763 (comment) <https://github.com/ceph/ceph/pull/62763#discussion_r2037882763> is actually a broken link, the anchor does not exist and following the link results in top of the document instead of the relevant section/anchor cloud-storage-class-configuration.
> 
> Please note that I am by no means an expert in RST, I may get terminology and details wrong but I'm basing this mostly on the Sphinx documentation below and how the files in doc/ usually do things.
> 
> Indeed links should be relative.
> 
> I meant RST references like the ones listed here in the documentation: https://www.sphinx-doc.org/en/master/usage/referencing.html#cross-referencing-arbitrary-locations
> 
> So there are a few different ways to create hyperlinks, I will list here 3 using RST features and 1 using hardcoded URLs (whether absolute or relative).
> Using RST references will create automatic link labels, for example if I add a reference just preceding a section with title Multisite Configuration then the link to that reference will automatically get the correct link text/label of "Multisite Configuration".
> Or I could directly link to a section using the section title text, without adding a reference in the destination document but this would break anchors in links if the section title is ever changed.
> 
> Linking to an RST reference when a specific section in the middle a document needs to be linked (like in this case the intended rendered link ends in an anchor), example:
> 
> See `Multisite Configuration`_ for multisite configuration instructions.
> This Multisite Configuration could be an RST item such as section title text or a custom defined anchor such as:
> 
> .. _Multisite Configuration: ../multisite
> Linking to an RST reference similarly as above but with different syntax, I believe this is about the same as the option (1) above:
> 
> :ref:`ceph-volume-lvm`
> And the reference that the hyperlink points to is defined similarly as same as above option (1). For example here the above link to reference ceph-volume-lvm will automatically get the link text/label from the section title ("lvm"):
> 
> doc/ceph-volume/lvm/index.rst:.. _ceph-volume-lvm:
> doc/ceph-volume/lvm/index.rst-
> doc/ceph-volume/lvm/index.rst-``lvm``
> doc/ceph-volume/lvm/index.rst-=======
> Linking a document name without an anchor in the intended link, in effect linking to the beginning of the document:
> 
> :doc:`cloud-transition`
> I believe this would be discouraged, a hardcoded link using HTML feature to an URL whether relative or absolute:
> 
> `upmap <../rados/operations/upmap>`_
> Some statistics from the doc/ code base:
> 
> 28k RST references of the above variety (1).
> 631 RST references of the above variety (2)
> 162 RST references of the above variety (3).
> 120 hardcoded relative links of the above variety (4)
> Almost all of them point to "special" content such as man pages and changelog .txt files, and not to RST files.
> Only 15 hardcoded absolute links of the above variety (4) to docs.ceph.com.
> These statistics point to RST references being the preferred method of linking other RST content, which seems logical. These are metadata inside the documentation and not tied to structure on the file system or the resulting URLs as rendered for the web server that may be changed over time resulting in broken links (I believe these are not linted to detect such cases??).
> 
> $ fgrep '`_' -R doc/ |wc -l
> 28767
> $ fgrep ':ref:`' -R doc/ |wc -l
> 631
> $ fgrep ':doc:`' -R doc/ |wc -l
> 162
> $ fgrep '<../' -R doc/ |wc -l
> 120
> $ fgrep '<https://docs.ceph.com' -R doc/ |wc -l
> 15
> —
> Reply to this email directly, view it on GitHub <https://github.com/ceph/ceph/pull/62763#issuecomment-2796187940>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTVONEI3E4UPG3EKVN4XYT2Y53LJAVCNFSM6AAAAAB23JJ2HSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJWGE4DOOJUGA>.
> You are receiving this because your review was requested.
>

anthonyeleven commented:
Not sure I see a reason to discourage #4

anthonyeleven commented:
If the rendering in the checks looks good, I think we're okay to merge?

zdover23 commented:
> If the rendering in the checks looks good, I think we're okay to merge?

I'm going to merge and backport this to S and to R.

zdover23 commented:
https://github.com/ceph/ceph/pull/62856 - Squid backport
https://github.com/ceph/ceph/pull/62857 - Reef backport
